File: java/src/main/java/sandbox/ObjectSerialize.java
package sandbox;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public class ObjectSerialize {

    public static Object deserialize(byte[] bytes) {
        Object result = null;
        try {
            ObjectInputStream oos =
                new ObjectInputStream(new ByteArrayInputStream(bytes));
            result = oos.readObject();
            oos.close();
        } catch (Exception ignore) { }
        return result;
    }

    public static byte[] serialize(Object o) {
        byte[] result = null;
        try {
            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            ObjectOutputStream oos = new ObjectOutputStream(baos);
            oos.writeObject(o);
            oos.flush();
            result = baos.toByteArray();
            oos.close();
        } catch (Exception ignore) { }
        return result;
    }

    public static Object deepClone(Object o) {
        return deserialize(serialize(o));
    }

}

File: junk/object-serialize.rb
java_import "java.io.ByteArrayInputStream"
java_import "java.io.ByteArrayOutputStream"
java_import "java.io.ObjectInputStream"
java_import "java.io.ObjectOutputStream"

def serialize(o)
  baos = ByteArrayOutputStream.new
  oos = ObjectOutputStream.new(baos)
  oos.writeObject(o)
  oos.flush()
  result = baos.toByteArray()
  oos.close
  result
end

def deserialize(bytes)
  ois = ObjectInputStream.new(ByteArrayInputStream.new(bytes))
  o = ois.readObject()
  ois.close
  o
end
